### DISCUSSION TITLE
Creating static compound body

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#117] [Q&A] [answered]
I am creating a body with static compound shape. The process is:
```js
const settings = new Jolt.StaticCompoundShapeSettings();
settings.AddShape(vec, quat, childSettings1);
settings.AddShape(vec, quat, childSettings2);
```

I was thinking to set position/rotation to zero/unit vectors and control the child offsets via `RotatedTranslatedShapeSettings` shape:
```js
const childSettings = new Jolt.RotatedTranslatedShapeSettings(vec, quat, shapeSettings);
```

However, it doesn't seem the compound shapes are correctly placed then. Specifically, it looks like only one of the children shapes is used.

**Comment by LeXXik** [comment]
I will use a non-rotated/translated shape and provide local position/rotation offsets to `RotatedTranslatedShapeSettings`, as that seems to work correctly. I will try to make a repro some time later this week.

**Comment by jrouwe** [comment]
> However, it doesn't seem the compound shapes are correctly placed then. Specifically, it looks like only one of the children shapes is used.

If you use only 1 childSettings object and repeatedly call ```Create()``` on it, it will actually only create a shape on the very first call, the next time it will return the same shape, regardless if you made any changes to the settings object. Is that maybe what's going wrong? (take a look at the recently added ```ClearCachedResult()```)

B.t.w. I would not recommend adding a ```RotatedTranslatedShape``` in a compound as it is a waste of CPU cycles. You will transform the shape twice on every collision query (even though the first transform is an identity transform).

**Reply by LeXXik** [reply]
Thank you for the hint on wasting the CPU cycles! Adjusted the logic to make use of it.

The issue with the missing shapes was in my setup, which auto-wrapped the shape settings into a scaled settings, if the shape scale has a non-unit scale, e.g.

```js
const shapeSettings = new Jolt.StaticCompoundShapeSettings();
shapeSettings.AddShape(zero, identity, 
	new Jolt.ScaledShapeSettings(
		new Jolt.RotatedTranslatedShapeSettings(jv, jq, new Jolt.SphereShapeSettings(r2)), 
		new Jolt.Vec3(0.4, 0.4, 0.4)
	)
);
```

Is this supported?

**Reply by LeXXik** [reply]
Tried to replicate in the "Falling Shapes" example. It looks like if I use scale that is >= 1, then the child shape appears (e.g. [2, 2, 2]). But if it is < 1 (e.g. [0.5, 0.5, 0.5]), then the shape doesn't appear.

**Reply by LeXXik** [reply]
Nevermind, the sphere simply becomes so small that it appears inside the connecting capsule. Its actually there. Hmm. I don't reuse childSettings, each one is newed. I will debug it some more.

**Reply by LeXXik** [reply, marked as answer]
Its a user error. Had an issue in my own code.
